theano/tensor/tests/test_opt.py

```diff
@@ -6498,6 +6498,46 @@ def test_local_merge_alloc():
     assert_raises((AssertionError, ValueError), f, 0., 1, 2, 5, 3, 4)
 
 
+def test_local_log_sum_exp1():
+    # Tests if optimization is applied
+    x = matrix('x')
+    y = T.log(T.sum(T.exp(x), axis=1))
+    MODE = theano.compile.get_default_mode().including('local_log_sum_exp')
+    f = function([x], y, mode=MODE)
+
+    assert (theano.scalar.basic.maximum
+            in [node.op.scalar_op for node in f.maker.fgraph.toposort()])
+
+
+def test_local_log_sum_exp2():
+    # Tests if the optimization works (result is correct) around 1.0
```

Comment:
test1 just checks that the optimization has been applied (if the opt is applied, then the graph should contain a maximum node).